src/webgpu/api/validation/render_pass/attachment_compatibility.spec.ts

```diff
@@ -182,6 +184,14 @@ g.test('render_pass_and_bundle,color_format')
     const { passFormat, bundleFormat } = t.params;
 
     t.skipIfTextureFormatNotSupported(passFormat, bundleFormat);
+    // float16 and float32 might miss color renderablity under compat mode
+    t.skipIf(
```

Comment:
would this be better to add something like

`t.selectDeviceToMakeFormatRenderableOrSkipTest`?

Then you could do

```js
t.selectDeviceToMakeFormatRenderableOrSkipTest(passFormat);
t.selectDeviceToMakeFormatRenderableOrSkipTest(bundleFormat);
```

Reply:
passFormat and bundleFormat are subcases param so we cannot call selectDevice* inside beforeAllSubcases?

Do you think it worth making passFormat and bundleFormat from subcases into cases for performance reason?

Reply:
Changing `paramsSubcaseOnly` to `params` seem to explode the number of params

Reply:
that's a good point. It would still be nice bundle this check into something like

`t.skipIfFormatNotRenderable(passFormat, bundleFormat)` right? or no? And it could check if the device has the feature.

Reply:
Yes! and fixed.